### PR TITLE
Add center of force calculations

### PR DESCRIPTION
**Pull request by anilyil**
## Purpose

This PR adds the center of force computations as ADflow cost functions. I am calling this center of force and not center of pressure because we look at the sum of all forces, which is what is more relevant for RANS solvers. 

The center of force cannot be exactly determined from just the total forces and moments; however, the total moments can be computed from the center of force and total force calculations. However, I kept the moment computations as is and did not modify them. This is because the code already had a very good and granular implementation for the moment terms. It is also a good sanity check to compare the relationships between moment, total force, and center of force. 

The center of force is returned in the following functionals:
```
cofxx
cofxy
cofxz
cofyx
cofyy
cofyz
cofzx
cofzy
cofzz
```
The first coordinate determines the force direction, and the second coordinate determines the component of the center of force. For example, `cofxy` is the y-coordinate of the center of x-force. It is up to the user how these are combined and used.

I also added a cost function called `mavgvi`, which is just a derived functional based on some compressible relationships. Not useful for most people.

## Expected time until merged
Similar to #246, after #231 is merged, I need to update this PR. Currently it looks like there are tons of changes but in reality this is a standalone feature. I also need to add tests.

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist

- [ ] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by lamkina — COMMENTED** [review]
~~Was there a reason you didn't add a test for the `mavgvi` cost function as well?~~
Nevermind since this shouldn't be used much.

**Review by lamkina — APPROVED** [review]
As far as I can tell, this looks good.

File: src/solver/surfaceIntegrations.F90 [hunk 1]
```diff
@@ -595,6 +675,32 @@ subroutine wallIntegrationFace(localValues, mm)
           Mv(2) = Mv(2) + my * blk
           Mv(3) = Mv(3) + mz * blk
 
+          ! the force integral for the center of pressure computation.
+          ! We need the cell centers wrt origin
+          xc = fourth*(xx(i,j,  1) + xx(i+1,j,  1) &
```
Comment (sseraj) [inline src/solver/surfaceIntegrations.F90:680]: I would changes these to `xco`, `yco`, `zco` as you call them in `flowIntegrationFace`. This helps to distinguish them from distances from the reference point.

File: src/solver/surfaceIntegrations.F90 [hunk 2]
```diff
@@ -449,6 +503,32 @@ subroutine wallIntegrationFace(localValues, mm)
...
+            +         xx(i,j+1,2) + xx(i+1,j+1,2))
+       zc = fourth*(xx(i,j,  3) + xx(i+1,j,  3) &
+            +         xx(i,j+1,3) + xx(i+1,j+1,3))
+
+       ! accumulate in the sums. each force component is tracked separately
+
+       ! Force-X
+       COFSumFx(1) = COFSumFx(1) + xc * fx * blk
+       COFSumFx(2) = COFSumFx(2) + yc * fx * blk
+       COFSumFx(3) = COFSumFx(3) + zc * fx * blk
```
Comment (sseraj) [inline src/solver/surfaceIntegrations.F90:520]: Do similar calculations also have to be added to zipperIntegrations.F90? The center of force test is nice, but it does not test the zipper integrations.

**Review by sseraj — COMMENTED** [review]
Looks pretty good, but I have a few comments

File: tests/reg_tests/refs/actuator_tests.json [hunk 3]
```diff
@@ -121,12 +135,13 @@
...
         "equationType": "Euler",
         "eulerWallTreatment": "linear pressure extrapolation",
+        "explicitSurfaceCallback": null,
         "firstRun": true,
         "flowType": "external",
         "forcesAsTractions": true,
         "frozenTurbulence": false,
         "globalPreconditioner": "additive Schwarz",
-        "gridFile": "input_files/actuator_test_pipe.cgns",
+        "gridFile": "tests/reg_tests/tests/reg_tests/tests/reg_tests/tests/reg_tests/tests/reg_tests/input_files/actuator_test_pipe.cgns",
```
[the remark this answers is not in the document]
Comment (sseraj) [reply tests/reg_tests/refs/actuator_tests.json]: I think this was the only file with this issue.

**Review by sseraj — COMMENTED** [review]
Looks good other than the flake8 check

**Review by sseraj — APPROVED** [review]

**Review by lamkina — APPROVED** [review]

**Pull request merged**